optuna/storages/base.py

```diff
@@ -95,6 +95,12 @@ def create_new_trial_id(self, study_id):
 
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def create_new_trial_number(self, trial_id):
```

Comment:
This method can be private and be implemented only in `RDBStorage`, because it's not accessed outside the storage.

Reply:
That sounds good.
`create_new_trial_number` is a workaround for the current `RDBStorage`, and it will be removed in the future.
Thank you for pointing it out.